File: dotnet/DesignPatterns/BehavioralDesignPatterns/VisitorDesignPattern/VisitorDesignPattern.Example1/ClothingProduct.cs
namespace VisitorDesignPattern.Example1;

/// <summary>
/// Concrete element: represents a clothing product.
/// Has an additional property (Size) specific to this type.
/// </summary>
public class ClothingProduct : IProduct
{
    public string Name { get; }
    public decimal Price { get; }
    public string Size { get; }

    public ClothingProduct(string name, decimal price, string size)
    {
        Name = name;
        Price = price;
        Size = size;
    }

    public void Accept(IVisitor visitor)
    {
        visitor.Visit(this);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/VisitorDesignPattern/VisitorDesignPattern.Example1/src/ClothingProduct.ts
import { IProduct } from './IProduct';
import { IVisitor } from './IVisitor';

export class ClothingProduct implements IProduct {
  readonly name: string;
  readonly price: number;
  readonly size: string;

  constructor(name: string, price: number, size: string) {
    this.name = name;
    this.price = price;
    this.size = size;
  }

  accept(visitor: IVisitor): void {
    visitor.visitClothing(this);
  }
}
